presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -361,8 +384,9 @@ private void invokeNoMoreSplitsIfNecessary()
                 schedulerUsesHostAddresses);
 
         // To support custom input formats, we want to call getSplits()
-        // on the input format to obtain file splits.
-        if (shouldUseFileSplitsFromInputFormat(inputFormat)) {
+        // on the input format to obtain file splits. CustomInputFormats can choose to either implement the annotation
+        // 'UseFileSplitsFromInputFormat' OR provide a PathFilter implementation for getting splits.
+        if (shouldUseFileSplitsFromInputFormat(inputFormat) && !inputPathFilter.isPresent()) {
```

Comment:
For my own understanding: so this means `inputPathFilter` would override `shouldUseFileSplitsFromInputFormat`, does my understanding sound right?

Reply:
Yes thats right.

Reply:
It is not as obvious from config perspective so let's make sure this is included in comment and also in the config description :)

Reply:
Makes sense. Will do.